### PR TITLE
Bump bitcoin-core version to v25.0

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/33513

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by goodov — APPROVED** [review]
`deps` lgtm

File: third_party/bitcoin-core/BUILD.gn [hunk 1]
```diff
@@ -109,8 +109,10 @@ static_library("secp256k1") {
     "src/src/secp256k1/src/group_impl.h",
     "src/src/secp256k1/src/hash.h",
     "src/src/secp256k1/src/hash_impl.h",
-    "src/src/secp256k1/src/num.h",
-    "src/src/secp256k1/src/num_impl.h",
+    "src/src/secp256k1/src/precomputed_ecmult.c",
```
Comment (kdenhartog) [inline third_party/bitcoin-core/BUILD.gn:112]: What's the motivation for this addition in this PR? Is it just a light refactor?

**Review by kdenhartog — APPROVED** [review]
No blocking concerns for me, so going to approve. I did have one question about the new precomputed_* files being added since they're not relevant to this change, but I don't see any issue with adding them.

File: third_party/bitcoin-core/BUILD.gn (hunk 1 above)
Comment (supermassive) [reply third_party/bitcoin-core/BUILD.gn:112]: That was some refactoring on bitcoin-core / secp256k1 side, so now we should adjust set of flies we include and link with.
For example precomputed_ecmult_gen.h appeared [there](https://github.com/bitcoin/bitcoin/commit/afb7a6fe06a33956ef43429d31f5934448f6e671#diff-315e82b72431171503317c60100564957d5e3b57b03b95932f8be806763d81d1)

**Review by yrliou — APPROVED** [review]
lgtm

**Pull request merged**